lib/associations/associations.rb

```diff
@@ -8,14 +8,16 @@ def self.extended(base)
 
       def has_many(association_id, scope = nil, **options, &extension)
         if options[:through]
-          klass_name = association_id.to_s.classify
-          klass = klass_name.safe_constantize
+          source_association_name = options[:source]&.to_s || association_id.to_s.singularize
```

Comment:
not sure why, but I do feel this is much easier to read.
If you are having trouble with it, you can add back.